src/Symfony/Component/VarDumper/Caster/DateCaster.php

```diff
@@ -68,4 +70,21 @@ private static function formatInterval(\DateInterval $i)
 
         return $i->format(rtrim($format));
     }
+
+    public static function castTimeZone(\DateTimeZone $timeZone, array $a, Stub $stub, $isNested, $filter)
+    {
+        $title = ($location = $timeZone->getLocation()) && extension_loaded('intl')
+            ? \Locale::getDisplayRegion('-'.$location['country_code'], \Locale::getDefault())
+            : ''
+        ;
+
+        $z = array(Caster::PREFIX_VIRTUAL.'timezone' => new ConstStub(self::formatTimeZone($timeZone), $title));
+
+        return $filter & Caster::EXCLUDE_VERBOSE ? $z : $z + $a;
+    }
+
+    private static function formatTimeZone(\DateTimeZone $z)
+    {
+        return (new \Datetime('now', $z))->format($z->getLocation() ? 'e (P)' : 'P');
```

Comment:
this method should be removed, it forces repeated calls to getLocation that inlinfing prevented - I'm sure things will be easier to read when removed. That means all changes on the castDateTime method should be reverted to me.